hack/update-api-reference-docs.sh

```diff
@@ -22,27 +22,74 @@ set -o nounset
 set -o pipefail
 
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
-REPO_DIR=${REPO_DIR:-"${PWD}/${KUBE_ROOT}"}
-DEFAULT_OUTPUT_PATH="${REPO_DIR}/docs/api-reference"
-OUTPUT=${1:-${DEFAULT_OUTPUT_PATH}}
+source "${KUBE_ROOT}/hack/lib/init.sh"
+kube::golang::setup_env
 
-echo "Generating api reference docs at ${OUTPUT}"
+DEFAULT_OUTPUT="${KUBE_ROOT}/docs/api-reference"
+OUTPUT=${1:-${DEFAULT_OUTPUT}}
+# Use REPO_DIR if provided so we can set it to the host-resolvable path
+# to the repo root if we are running this script from a container with
+# docker mounted in as a volume.
+# We pass the host output dir as the source dir to `docker run -v`, but use
+# the regular one to compute diff (they will be the same if running this
+# test on the host, potentially different if running in a container).
+REPO_DIR=${REPO_DIR:-"${KUBE_ROOT}"}
```

Comment:
@jlowdermilk I moved the reference to REPO_DIR from the verify script to the update script.